### ISSUE TITLE
Caret is hidden when opening note in vim from CLI tool

### ISSUE THREAD

**Issue by createchange** [laurent22/joplin#147]
### Operating system

- [ ] Windows
- [ ] macOS
- [X] Linux (Arch), node v9.3.0
- [ ] Android
- [X] iOS

### Application

- [ ] Desktop
- [X] Mobile
- [X] Terminal

*If there is an error, please copy and paste the full error message*
Arch Linux CLI - Error Message: Unexpected token o in JSON at position 1
---
I am also having issues with syncing on my iOS device, so I am assuming the issue is somewhere on my end. I did authorize OneDrive (new account) on both CLI and iOS devices (both fresh installs). 

I am happy to provide other information that you might need, but may need some in providing some of that information - working on my Linux-fu by the day, but am likely going to be out of my element here.

**Comment by laurent22** [comment]
Hello, there was a bug in the CLI app, which I've now re-published. Please could you try again? Also please wait for the next iOS version 0.10.9 (should be available today or tomorrow) and try with this one.

**Comment by createchange** [comment]
Thanks for the response. I was able to install the new version, but Microsoft decided to suspend my newly created account for suspicious activity. Not sure if this is tied to Joplin (it is a new account, so possibly? or maybe it was because I tried syncing a million times while it was broken?) but the sooner you get Joplin working with Nextcloud, the better! When my account is unblocked, if ever, I will give this another shot. Honestly though, I'd prefer not to have any accounts with MS, but this app fills a need for me.

Thanks for your hard work. I'll update this when possible.

Take care.

**Comment by createchange** [comment]
Laurent,
Problem was solved on both iOS and Linux.
Thanks for your work!

**Comment by createchange** [comment]
Laurent - do you have any way to contact you for questions regarding Joplin that are not bugs? I'll just mention my specific question here: when editing a note, vim opens up but my carat block is not visible, so I just kind of trial and error where I am going to be entering text at times.

**Comment by laurent22** [comment]
A fix for this has actually been suggested but I haven't had time to add it yet: https://github.com/cronvel/terminal-kit/issues/50#issuecomment-348893878

I'm re-opening the issue for now to keep track of it.

**Comment by laurent22** [comment]
@createchange, actually I cannot replicate this bug, does it happen every time or is it randomly? By the way which Linux distribution is it?

**Comment by createchange** [comment]
@laurent22, thanks for following up. It happens every time for me. I am using Arch. Please let me know if there is any other information I can provide you and I’d be glad to.

**Comment by laurent22** [comment]
I'm not familiar with Arch Linux, is there any easy to use installer like for Ubuntu? I've tried the ISO they provide in VirtualBox but got to a prompt where I guess I'd have to install everything manually? Is there any quick way to do this?

**Comment by createchange** [comment]
There’s really not a easy graphical install option for Arch. There were a few projects that added a GUI to facilitate the process, but I just looked at them and they seem abandoned.

The best bet I think would be Antergos (https://antergos.com/), which is a derivative based on Arch.

**Comment by laurent22** [comment]
I've tried with Antergos and it's working for me (Caret is visible in vim). Which terminal emulator, and which version of Bash are you using?

**Comment by createchange** [comment]
Good call on that. I used zsh, not bash, and I think that is where the problem lies. If I change to my root user, which uses bash as the default shell, the caret shows while editing a note in joplin. It appears that the problem is that I am using zsh. Do you know of any solution to this, now that we have it narrowed down?

**Comment by laurent22** [comment]
Unfortunately I still cannot replicate this in neither zsh under WSL nor zsh under Antergos. Does it work for you with a default vim installation (i.e. with no plugin, no vimrc, etc.)? Does it work with a default zsh installation? I guess there's some plugin or setting that's causing this but not sure what.

**Comment by createchange** [comment]
You nailed it. I had "set term=ansi" in my .vimrc, and removing that fixed the problem. Not only that, but it made my vim look better. so double win! Now to open a new issue... You're doing great work.

**Comment by laurent22** [comment]
Ok good to know the reason for this bug. I'll add this to a faq.
